src/libertem/io/dataset/ser.py

```diff
@@ -217,29 +265,50 @@ def _preprocess(self, tile_data, tile_slice):
         self._corrections.apply(tile_data, tile_slice)
 
     def get_tiles(self, tiling_scheme, dest_dtype="float32", roi=None):
+        sync_offset = self.meta.sync_offset
         shape = Shape((1,) + tuple(self.shape.sig), sig_dims=self.shape.sig.dims)
 
         self.validate_tiling_scheme(tiling_scheme)
 
         start = self._start_frame
-        stop = start + self._num_frames
-        if roi is None:
-            indices = np.arange(start, stop)
-            offset = start
-        else:
-            indices = _roi_to_indices(roi, start, stop)
-            offset = np.count_nonzero(roi.reshape((-1,))[:start])
-
-        with fileSER(self._path) as f:
-            for num, idx in enumerate(indices):
-                tile_slice = Slice(origin=(num + offset, 0, 0), shape=shape)
-                data, metadata = f.getDataset(int(idx))
-                if data.dtype != np.dtype(dest_dtype):
-                    data = data.astype(dest_dtype)
-                data = data.reshape(shape)
-                self._preprocess(data, tile_slice)
-                yield DataTile(
-                    data,
-                    tile_slice=tile_slice,
-                    scheme_idx=0,
-                )
+        if start < self.meta.image_count:
+            stop = min(start + self._num_frames, self.meta.image_count)
+            if roi is None:
+                indices = np.arange(start, stop)
+                if sync_offset < 0:
+                    indices[indices < 0] = 0
+                offset = start - sync_offset
+            else:
+                indices = _roi_to_indices(roi, start, stop, sync_offset)
+                offset = np.count_nonzero(roi.reshape((-1,))[:start - sync_offset])
+
+            with fileSER(self._path) as f:
+                for num, idx in enumerate(indices):
+                    origin = (num + offset,) + tuple([0] * self.shape.sig.dims)
+                    if sync_offset < 0 and roi is None:
+                        if abs(sync_offset) > (origin[0] + shape[0]):
+                            continue
+                    tile_slice = Slice(origin=origin, shape=shape)
+                    data, metadata = f.getDataset(int(idx))
+                    if data.dtype != np.dtype(dest_dtype):
+                        data = data.astype(dest_dtype)
+                    data = data.reshape(shape)
+                    self._preprocess(data, tile_slice)
+                    if sync_offset < 0 and roi is None:
+                        if abs(sync_offset) > origin[0]:
```

Comment:
Maybe should add some comments here describing the condition you are actually testing for here, as I don't think it is intuitively clear from reading the code